SigmaClientTool/Main.cpp

```diff
@@ -0,0 +1,143 @@
+/*Copyright 2017 Bang & Olufsen A/S
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.*/
+
+#include "SigmaTcpClient.h"
+#include <iostream>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h> 
+
+void PrintUsage()
+{
+	std::cout << "Usage:" << std::endl << "Read: SigmaClientTool [target IP] [read_int|read_dec|read_reg] [address] [length] (for read_reg only)" \
+		<< std::endl << "Write value: SigmaClientTool [target IP] [write_val] [address] [value] (single int or decimal value)" \
+		<< std::endl << "Write register: SigmaClientTool [target IP] [write_reg] [address] [length] [value1 value2 value3 ...] (each value is 1 bytes in hex format 0xXX)" <<std::endl;
+}
+
+int main(int argc, char* argv[])
+{
+
+	if (argc < 3) {
+		PrintUsage();
+		return 0;
+	}
+
+	SigmaTcpClient tcpClient;
+	std::cout << "Initializing connection to " << argv[1] << std::endl; 
+	tcpClient.InitializeConnection(argv[1]);
+
+	if (strcmp(argv[2], "read_reg") == 0)
+	{
+		if (argc < 5) {
+			PrintUsage();
+			return 0;
+		}
+
+		std::cout << "Reading " << argv[4] << " bytes from " << argv[3] << std::endl;
+
+		SigmaReadResponse& resp = tcpClient.ReadMemory((uint16_t)strtol(argv[3], nullptr, 0), (uint16_t)strtol(argv[4], nullptr, 0));
+		printf("0: ");
+		for (int i = 0; i < resp.length; i++)
+		{
+			printf("0x%02x", resp.data[i]);
+			(i % 12 == 0 && i != 0) ? printf("\n%i: ", i) : printf(" ");
+		}
+		printf("\n");
+	}
+	else if (strcmp(argv[2], "read_int") == 0 || strcmp(argv[2], "read_dec") == 0)
```

Comment:
Instead of using numbers that is not explained maybe us the const in h-file or macro in the same way that you have used const char CommandRead. It makes the code nicer and easier to understand.